File: src/main/java/corp/soyummy/ingredients/IngredientRepository.java
package corp.soyummy.ingredients;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

@Repository
public interface IngredientRepository extends MongoRepository<Ingredient, String> {
    List<Ingredient> findByTitleContainingIgnoreCase(String title);
}

File: src/main/kotlin/com/soyummy/ingredients/IngredientRepository.kt
package com.soyummy.ingredients

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

@Repository
interface IngredientRepository : MongoRepository<Ingredient, String> {
    fun findByTitleContainingIgnoreCase(title: String, pageable: Pageable): Page<Ingredient>
}